api/room/schema.py

```diff
@@ -49,13 +52,33 @@ def mutate(self, info, office_id, **kwargs):
         get_office = Office.query.filter_by(id=office_id).first()
         if not get_office:
             raise GraphQLError("No Office Found")
-
+        if 'block_id' in kwargs:
+            exact_block = Block.query.filter_by(id=kwargs['block_id']).first()
+            if not exact_block:
+                raise GraphQLError("Block with such id does not exist")
+            floor = Floor.query.filter_by(
+                block_id=exact_block.office_id).first()
+            if floor.id == kwargs['floor_id']:
```

Comment:
A Block can have multiple floors, but from this query(line 59) we only return the first floor instead of a list of floors. Therefore if the `floor_id` for an existing floor is passed, but doesn’t match the one returned from the Floor query, the room won’t be registered.

Reply:
Yes @peterwade153, the room should be created in the floor whose `floor id` is passed, so we query if that floor exists in that particular block whose `block id` is also passed then we create the room, otherwise, we don't create the room having in mind that `floor id` is unique for each floor.

Reply:
I agree a room is only created if the block and floor both exist, 👍 @Bonifase